Feature/nex 150 update code from old repo

btamas opened the pull request:
Jira: https://oat-sa.atlassian.net/browse/NEX-150

Update code from `extension-tao-tests`

Test: `npm run test`
Test new ES6 coverage measurement: `npm run test:cov` then `npm run coverage:html`

Related PR:
- [ ] https://github.com/oat-sa/extension-tao-test/pull/290

Task after merge:
- [x] Release it to npm

btamas committed:
update code and tests

btamas committed:
add json loader for requirejs

btamas committed:
update modules

btamas committed:
use rollup istanbul instrumenter

btamas committed:
version bump

ferenckiss89 commented on src/runnerComponent.js:
```diff
@@ -17,148 +17,135 @@
+        return config.provider[type];
+    }
+
+    if (config.providers && config.providers[type]) {
```
if (!config.providers || !config.providers[type]) {
    return false;
}
const typeProviders = config.providers[type];
if (typeof typeProviders === 'object' && (typeProviders.id || typeProviders.name)) {
    return typeProviders.id || typeProviders.name;
}
if (Array.isArray(typeProviders) && typeProviders.length > 0) {
    return typeProviders[0].id || typeProviders[0].name;
}

ferenckiss89 commented on test/mocks/mockAltRunnerProvider.js:
```diff
@@ -0,0 +1,13 @@
+define([], function () {
```
license header is missing

ferenckiss89 commented on test/mocks/mockInvalidRunnerProvider.js:
```diff
@@ -0,0 +1,7 @@
+define([], function () {
```
missing license header

ferenckiss89 commented on test/mocks/mockPlugin2.js:
```diff
@@ -1,16 +1,13 @@
-define([], function () {
```
missing license header

ferenckiss89 commented on test/mocks/mockProxyProvider.js:
```diff
@@ -0,0 +1,8 @@
+define([], function () {
```
missing license header

ferenckiss89 commented on test/mocks/mockRunnerProvider.js:
```diff
@@ -0,0 +1,13 @@
+define([], function () {
```
missing license header

ferenckiss89 reviewed (commented):
First round is done on my side, only code style check and test running. Little comments.

btamas committed:
add license headers

ferenckiss89 reviewed (approved):
Ok for me.

jsconan commented on test/providerLoader/test.html:
```diff
@@ -0,0 +1,21 @@
+<html>
+    <head>
+        <meta charset="utf-8" />
+        <title>Test Runner - Runner Component</title>
```
Wrong title, should be `Test Runner - Provider Loader`

btamas replied:
I just copied the file, but I totally agree. Fixed.

jsconan reviewed (commented):
Looks good.
Only have one remark, nothing blocking.

I also noticed some new unit tests are converted to ES6, while the majority is ES5. That should not be a big issue, but would prevent to run those tests on older browsers (only source code is transpiled, not tests). Anyway, this is not something to change, just a wondering.

btamas committed:
fix test title

btamas commented:
> Looks good.
> Only have one remark, nothing blocking.
> 
> I also noticed some new unit tests are converted to ES6, while the majority is ES5. That should not be a big issue, but would prevent to run those tests on older browsers (only source code is transpiled, not tests). Anyway, this is not something to change, just a wondering.

Based on @krampstudio 's presentation last week, we can use ES6 in tests with knowing it won't work in old browsers. It is an acceptable side effect.
The reason why I modified only some part of the tests, because I fixed `eslint` errors what was easy to fix and I did not want to spend time to refactor everything, because it was not part of this task.

jsconan reviewed (approved):
A little typo in the changed title, but anyway :)
